src/plugins/data_views/common/data_views/data_views.ts

```diff
@@ -939,21 +939,20 @@ export class DataViewsService {
     skipFetchFields = false,
     displayErrors = true
   ): Promise<DataView> {
+    const doCreate = () => this.createFromSpec(spec, skipFetchFields, displayErrors);
+
     if (spec.id) {
-      const cachedDataView = spec.id ? await this.dataViewCache.get(spec.id) : undefined;
+      const cachedDataView = await this.dataViewCache.get(spec.id);
```

Comment:
this doesn't work if there are multiple calls to this method from the same call stack because it's awaiting the promise from the cache which is waiting for the next tick. Removing the `await` should be enough - this returns the promise in the cache right away instead of waiting for it to resolve.